File: src/main/java/com/inner_circle/survey/repository/QuestionRepository.java
package com.inner_circle.survey.repository;

import com.inner_circle.survey.entity.request.Question;
import com.inner_circle.survey.entity.request.Survey;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface QuestionRepository extends JpaRepository<Question, Long> {

  List<Question> findBySurveyAndLatestTrue(Survey survey);
}

File: src/main/kotlin/org/inner/circle/survey/infrastructure/repository/QuestionRepository.kt
package org.inner.circle.survey.infrastructure.repository

import org.inner.circle.survey.domain.entity.Question
import org.inner.circle.survey.domain.entity.Survey
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface QuestionRepository : JpaRepository<Question, Long> {
    fun findBySurvey(survey: Survey): List<Question>

    fun findByIdAndSurvey(
        questionId: Long,
        survey: Survey,
    ): Question?
}
